### PR TITLE
[IMP] util/records.py: allow to add qweb view

### PR DESCRIPTION
**Pull request by hpr-odoo**
logic of key [generation](https://github.com/odoo/odoo/blob/8f72c83168b70eb9ca3b1829de91163ee52003d8/odoo/addons/base/models/ir_ui_view.py#L489)

### CODE REVIEW

File: src/util/records.py [hunk 1]
```diff
@@ -272,24 +273,28 @@ def get_trans_terms(value):
...
     arch_column_value = Json({"en_US": arch_db}) if jsonb_column else arch_db
     cr.execute(
         """
         INSERT INTO ir_ui_view(name, "type",  model, inherit_id, mode, active, priority, %s)
-        VALUES(%%(name)s, %%(view_type)s, %%(model)s, %%(inherit_id)s, %%(mode)s, 't', %%(priority)s, %%(arch_db)s)
+        VALUES(%%(name)s, %%(view_type)s, %%(model)s, %%(inherit_id)s, %%(mode)s, 't', %%(priority)s, %%(arch_db)s %s)
         RETURNING id
     """
-        % arch_col,
+        % (arch_col + (", key" if key_exist else ""), ", '%s'" % key if key_exist else ""),
```
Comment (KangOl) [inline src/util/records.py:297]: This is not readable :/

At least, don't inject an unsafe variable in the query.
```suggestion
        % (arch_col + (", key" if key_exist else ""), ", %(key)s" if key_exist else ""),
```

Comment (hpr-odoo) [reply src/util/records.py]: updated, Thank you for your input :slightly_smiling_face:

**Comment by hpr-odoo** [comment]
Should matt be green? not failing by this fix

**Comment by KangOl** [comment]
upgradeci retry with always only base

**Comment by KangOl** [comment]
upgradeci retry with always only base in all versions

File: src/util/records.py [hunk 2]
```diff
@@ -272,31 +273,36 @@ def get_trans_terms(value):
...
-        % arch_col,
+        % (arch_col + (", key" if key_exist else ""), ", %(key)s" if key_exist else ""),
         {
             "name": name,
             "view_type": view_type,
             "model": model,
             "inherit_id": inherit_id,
             "mode": "extension" if inherit_id else "primary",
             "priority": priority,
+            "key": key if key_exist else "",
```
Comment (KangOl) [inline src/util/records.py:305]:
```suggestion
            "key": key,
```

File: src/util/records.py [hunk 3]
```diff
@@ -272,31 +273,36 @@ def get_trans_terms(value):
             )
 
 
-def add_view(cr, name, model, view_type, arch_db, inherit_xml_id=None, priority=16):
+def add_view(cr, name, model, view_type, arch_db, inherit_xml_id=None, priority=16, key=""):
```
[the remark this answers is not in the document]
Comment (hpr-odoo) [reply src/util/records.py]: Done, Matt is happy now :slightly_smiling_face:

**Comment by KangOl** [comment]
@robodoo r+

**Pull request closed without merging**